### ISSUE TITLE
Proposal: new syntax for defining abstract methods

### ISSUE THREAD

**Issue by fphh** [Microsoft/TypeScript#6833]
Currently, we write
```

abstract class A {
  abstract f() : void;
} 
```

Wouldn't it be nicer to write:
```

abstract class A {
  abstract f : () => void;
}

```

That would syntactically go smoothly with type aliases:
```

type t = () => void;
abstract class A {
  abstract f : t;
}

```
which currently (1.7.5) is disallowed. (See #6832).

**Comment by sandersn** [comment]
That looks like an abstract property to me. Take a look at #4469 -- is that what you're looking for?

**Comment by fphh** [comment]
Sorry, no, I was talking about declaring the type of an abstract method. If we consider
```

type t = (x : number) => void;
abstract class {
   abstract f(x : number) : void;
}
```

then I would argue that _f_ has type _t_. So why can't we just write:

```

type t = (x : number) => void;
abstract class {
  abstract f : (x : number) => void;
}
```

Pro:
1. This unifies the syntax for abstract methods with the syntax for abstract properties.
2. If type aliases were fully implemented (#6832), we could write for short;

```

type t = (x : number) => void;
abstract class {
  abstract f : t;
}
```

**Comment by mhegazy** [comment]
The syntax you are proposing here is already used for property declarations in typescript. properties are not the same as methods in a class, methods go on the prototype, properties go on the instance, i do not think there is much we can change here.

today:

```ts
class C {
    a: T;   // this is a property whose name is "a" and type is T, and not a method
}
```

Consider using an interface instead:

```ts
type t = (x: number) => void;

interface A {
    f: t;
}

class Base implements A{
    f(x: number) {
    }
}
```
